File: 프로그래머스/0/120813. 짝수는 싫어요/짝수는 싫어요.java
class Solution {
    public int[] solution(int n) {
        int count = 1;
        
        for (int i = 1; i < n; i++) {
            if (i%2 != 0) {
                ++count;
            }
        }
        if (n % 2 == 0) {
            count--;
        }
        
        int answer[] = new int[count];
        count = 0;
        
        for (int i = 1; i <= n; i++) {
            if (i%2 != 0) {
                answer[count] = i;
                count++;
            }
        }
        return answer;
    }
}

File: 프로그래머스/0/120813. 짝수는 싫어요/짝수는 싫어요.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int n) {
    int num = 0; 
    if (n%2 == 0) {
        num = n/2;
    } else {
        num = n/2+1;
    }
    int* answer = (int*)malloc(num * sizeof(int));
    int count = 0; 
    
    for (int i = 0; i <= n; i++) {
        if (i%2 == 0) {
            continue;
        }
        answer[count] = i;
        count++;
    }
    return answer;
}